### ISSUE TITLE
<div> in <dl>

### ISSUE THREAD

**Issue by zcorpan** [w3c/editing#153]
See https://github.com/whatwg/html/commit/5454d702e0262749aba55576dda32c48e0e498f0

As far as editing goes, I suppose https://w3c.github.io/editing/execCommand.html is what is affected (e.g. "allowed child").

**Comment by zcorpan** [comment]
https://github.com/whatwg/html/issues/1937 is the relevant issue with more information about the change.

**Comment by johanneswilm** [comment]
@zcorpan Could you briefly state what change we have to do to the spec to be in compliance with what you decided upon, so that we don't have to read through the entire whatwg discussion thread?

**Comment by zcorpan** [comment]
Sure!

The content model of `dl` now allows either:
- 1+ `div`s (plus `script` and `template`)
- or 0+ groups of: 1+ `dt` elements followed by 1+ `dd` elements (plus `script` and `template`)

The content model of `div` in `dl` is: 1+ `dt` elements followed by 1+ `dd` elements (plus `script` and `template`).

See https://html.spec.whatwg.org/multipage/semantics.html#the-dl-element for the spec and examples.

**Comment by johanneswilm** [comment]
@zcorpan I quickly looked through the document you mention using the find function. In all but one case there was an even match between dt and dd elements. In one case there were two dt elements followed by one dd. I guess that's still legal, but it seemed like the first dt-element possibly was unrelated to the dd-element, so I inserted an empty dd-element. Is that all this was about, or is there something more I am missing?

**Comment by zcorpan** [comment]
A dt/dd group consists of **one or more** `dt` followed by **one or more** `dd`; you can have as many as you want of each. This has always been the case in HTML. What's new is the `div` element.

This issue is not about the source markup of `execCommand.html`; that is fine as it was (your change should be reverted).

This issue is about the normative text in that document. What should happen when pressing enter in `<dl><div><dt>x</dt><dd>y][</dd></div></dl>` for example.

**Comment by johanneswilm** [comment]
> A dt/dd group consists of **one or more** `dt` followed by **one or more** `dd`; you can have as many as you want of each. This has always been the case in HTML. What's new is the `div` element.

Right, but they should be related, no? Maybe they are, my scan of the document wasn't that great. I was just looking for something I thought you thought might be a bug.

Anyway, I reverted my change as this wasn't what your request was about, and we only work on the execCommand spec when it's really really needed. In general, everything is moving in other directions, at least for now.

> This issue is about the normative text in that document. What should happen when pressing enter in `<dl><div><dt>x</dt><dd>y][</dd></div></dl>` for example.

Ah, you did read the big disclaimer at the top of that text, right? contenteditable=true/execCommand does output a lot of invalid html.We have decided not to work on this right now as it's a monstrous task that will take decades to complete. You are not implementing anything from the execCommand spec right now, are you?

If you are working on changes to the DOM caused by editors, I think the best advice one can give is to create something or other that looks like valid HTML. 

You may also want to look at the input events spec instead, as in real-world applications browsers seldom are the ones deciding about such changes to the DOM structure. It's generally JS editors that decide how to respond to user input, and the beforeinput event helps them with that.

**Comment by zcorpan** [comment]
> We have decided not to work on this right now as it's a monstrous task that will take decades to complete.

OK, but this issue is still valid. If someone is to tweak how editing in `dl` behaves in a browser engine, they should be aware of this issue.

**Comment by johanneswilm** [comment]
They probably should be working on something else. 😊 seriously, it's a waste of time, as JS will override it anyway, and no editor will work without JS. 

But yes, the current draft basically says that "we permit some non-valid html but we only have a few random examples here of things we think should work". Maybe this should be changed to something along the lines of "the browser must guarantee to produce valid html at all times, please go and read in these other specs what valid html looks like right now".
But to define what exactly happens in every situation  is the mega task that we won't be able to do in our lifetime, most likely.
